From fd431e9639adde55cbc1b05861d66f80d92faf22 Mon Sep 17 00:00:00 2001
From: Joao Carreira <jfmcarreira@gmail.com>
Date: Fri, 9 May 2014 18:21:59 +0100
Subject: [PATCH] Add a new variable WITH_AMZ

---
 CMakeLists.txt | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1f66e16..0d4d041 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,6 +12,7 @@ option(WITH_MP3Tunes "Enable mp3tunes in the Amarok player, requires multiple ex
 option(WITH_IPOD "Enable iPod support in Amarok" ON)
 option(WITH_MYSQL_EMBEDDED "Build the embedded database library -- highly recommended" ON)
 option(WITH_PLAYGROUND "Enable building of playground scripts and applets (WARNING: some of them might have legal issues!)" OFF)
+option(WITH_AMZ "Enable AMZ downloader" ON)
 
 include(CheckLibraryExists)
 check_library_exists(dl dlopen "" LIBDL_FOUND)
@@ -219,10 +220,12 @@ if( WITH_PLAYER )
     find_package(GLIB2)
     macro_log_feature( GLIB2_FOUND "glib2" "Required by libgpod and mp3tunes" "http://www.gtk.org" FALSE "2.x" "")
     endif( WITH_IPOD OR WITH_MP3Tunes )
-
-    find_program( CLAMZ_FOUND clamz PATH )
-    macro_log_feature( CLAMZ_FOUND "clamz" "Optional requirement to download songs from the Amazon MP3 store. Highly recommended on Linux, as the official downloader from Amazon is quite broken on many systems." "https://code.google.com/p/clamz/" FALSE )
-
+    
+    if( WITH_AMZ )
+      find_program( CLAMZ_FOUND clamz PATH )
+      macro_log_feature( CLAMZ_FOUND "clamz" "Optional requirement to download songs from the Amazon MP3 store. Highly recommended on Linux, as the official downloader from Amazon is quite broken on many systems." "https://code.google.com/p/clamz/" FALSE )
+    endif( WITH_AMZ )
+    
     include_directories( ${KDE4_INCLUDES} )
 
     if( KDE4_BUILD_TESTS AND NOT WIN32 )
-- 
1.9.1

